### PR TITLE
feat(core): keyword form for log_prob-family ops via _pack_value (#228 PR 1)

### PR DESCRIPTION
**Pull request by jhuggins**
## Summary

Implements **PR 1 of #228** — the atomic core: a keyword form for the
`log_prob`-family ops, backed by a per-class `_pack_value` adapter, so users can
write

```python
log_prob(model, intercept=0.0, slope=0.5, X=X_obs, y=y_obs)
log_prob(prior, intercept=0.0, slope=0.5)
log_prob(Normal(name="x"), x=1.5)
```

instead of having to know each distribution's bespoke value layout. The
positional form is unchanged and still broadcasts.

Part of #228. (Remaining: PR 2 `condition_on` case-mismatch, PR 3 StanModel
Tier 2, PR 4 `RandomMeasure` kwarg form; follow-up #229.)

## What changed

- **Protocols genericized** — `SupportsLogProb[T]` / `SupportsUnnormalizedLogProb[T]`
  are now generic in the sample type `T`, with `_log_prob(value: T | ArrayLike)`.
  Documents the value contract; runtime-checkable behavior is unchanged.
- **`Distribution._pack_value(**kwargs) -> T`** (new, on the base) — builds one
  draw from named fields: single-field → bare value (`T = Array`); multi-field →
  `Record` in field order. Raises clearly on missing/unexpected fields.
- **Op layer rewritten as outer wrapper + inner `@workflow_function`** for
  `log_prob`, `prob`, `unnormalized_log_prob`, `unnormalized_prob`, and the two
  `random_*_log_prob` ops. The outer wrapper resolves the positional-or-keyword
  value and forwards the WF controls (`seed` / `n_broadcast_samples` /
  `include_inputs`) explicitly, so a field named like a reserved control can't be
  silently swallowed. Broadcasting still happens — the inner `_<op>_impl` is the
  WorkflowFunction.
- **Reserved-name warning** — constructing a distribution whose field collides
  with `{seed, n_broadcast_samples, include_inputs}` warns (cheap, side-effect-free
  check in the metaclass; points users to the positional `Record` escape hatch).
- **`SimpleModel._log_prob` accepts a `Record` *or* a `(params, data)` tuple.**
  The Record branch (kwarg form) splits into the prior's value (repacked via the
  prior's own `_pack_value`) and a data sub-record from the likelihood's
  `data_template`. See "Deviation" below.
- **`SequentialJointDistribution`** already handled `Record` at runtime
  (`_eval_log_prob` converts it); annotations tightened to say so.
- **`StanModel` / `_UnconstrainedStanView` Tier 1** — `_pack_value` accepts a
  single `parameters=` flat array (the form `_log_prob` already consumes). See
  "Not runtime-verified" below.

## Deviation from the issue plan (justified)

The plan said "tuple form *gone*" for `SimpleModel`. In practice a likelihood
without a `data_template` (bare-array data, e.g. the `GaussianLikelihood` test
fixture) has **no named data fields**, so its data can't be expressed as kwargs;
the `(params, data)` tuple is the only way. `SimpleModel._log_prob` therefore
**dual-accepts** Record and tuple. Consequences:

- The headline kwarg form works for likelihoods with a `data_template` (GLM).
- The tuple form remains valid, so the **"5 tuple-form call sites"** in the
  checklist need **no change** — `_inference_utils.py`'s bare-target path and the
  existing `test_simple_model.py` / `test_sbijax.py` tuple tests keep working.
  (`build_target_log_prob` already bypasses `SimpleModel._log_prob` for the
  structured MCMC path, so inference is unaffected either way.)

## Test plan

- **New:** `tests/core/test_log_prob_kwargs.py` — kwarg form across scalar dists
  (Normal/Beta/MVN), `ProductDistribution`, `JointGaussian`, `SimpleModel` (GLM,
  asserting kwarg == positional-Record == tuple), `StanModel._pack_value`
  (in isolation), error paths (missing / extra / positional+kwargs / neither),
  `prob` / `unnormalized_log_prob` kwarg forms, and the reserved-name warning.
- **Updated:** `test_ops.py` (ops are now thin wrappers over inner
  `_<op>_impl` WorkflowFunctions; positional-only `value`+kwargs collision);
  `test_distribution_array.py` (one call switched from keyword `value=` to
  positional — see Breaking changes).
- **Regression:** full `tests/{core,distributions,modeling,inference,converters}`
  run; my branch introduces **no new failures** vs `origin/main` (verified by
  diffing the full failure sets — the only failures are pre-existing
  Prefect-detection flakes in `test_workflow_parallel.py`, identical on both).

## Breaking changes

`log_prob` / `prob` / `unnormalized_log_prob` / `unnormalized_prob` /
`random_*_log_prob` now take `dist` and `value` as **positional-only** (so field
names are free for the keyword form). The only in-repo keyword-`value=` call site
was one test, updated to positional. `log_prob(dist, value)` (positional) is
unchanged. Labeled `kind:breaking-change`.

## Not runtime-verified

`StanModel` requires `bridgestan` (not installed in CI / locally), so its
`_pack_value` is tested in isolation via `object.__new__` (per STYLE_GUIDE §8.4)
but the full `log_prob(stan_model, parameters=...)` path is not exercised here.

## Documentation

- [x] Docstrings updated for changed public APIs (`log_prob` & friends,
  `_pack_value`, `SimpleModel._log_prob`, protocol contracts)
- [ ] User Guide / tutorials — none needed for this layer
- [x] CHANGELOG / release-notes — noted in #228

## Checklist

- [x] PR title follows `<type>(<scope>): <subject>`
- [x] At least one `area:*` label applied
- [x] Linked to an issue above

### CODE REVIEW

**Comment by jhuggins** [comment]
Superseded by #282 — same feature (keyword form for the `log_prob`-family ops, issue #228 PR-1), reimplemented cleanly:

- The ops stay **plain `WorkflowFunction`s** (no wrapper / `_DensityOp`), resolving the keyword form in their body via `Distribution._pack_value` — consistent with how `condition_on` handles named data kwargs.
- Branched off **current `main`** (includes #280), so the suite is fully green (0 failures, 17 skipped).
- Non-breaking: `value=` still works; the keyword form is purely additive.

This branch's history churned through `_DensityOp` (introduced → de-smelled → removed). Closing in favor of the clean #282, which carries the full feature.
